GameArena.java

```diff
@@ -458,6 +458,15 @@ public void removeText(Text t)
 		this.removeObject(t);
 	}
 
+	/**
+	 * Removes every object that has ever been added to the GameArena. Nothing
+	 * should appear on the GameArena window after this has executed.
+	 */
+	public void clearGameArena() {
+		things.clear();
```

Comment:
Looks good!

I think we should protect this code with a synchronized block though. This would prevent the list of objects being cleared during the time that the list is iterated over by the draw() method. I suspect we would get a crash if the user cleared the list during this time.

Could you add a synchronized block around this line, like we do when adding an object:
https://github.com/finneyj/GameArena/blob/master/GameArena.java#L311

If you can add that, and quickly test that it works OK on your code, I'll happily merge this in. :)